### PR TITLE
Tutorial for creating pages

### PR DESCRIPTION
**Pull request by brockfanning**
Q | A
--- | ---
Feature branch/test site URL | NA
Fixed issues | NA
Related version | 1.4.0-dev
Bugfix, feature or docs? | docs
Keeps backward-compatibility? |NA
Includes tests? |NA
Updated docs? |✔️
Updated config forms? |NA
Added CHANGELOG entry? |NA

### CODE REVIEW

File: docs/tutorials/create-pages.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+This creates a branch called "create-pages-tutorial" and switches to that branch. We will be working on this branch for the rest of the tutorial.
+
+## Start the new page
+
+1. In your site repository on Github.com, go into the `_pages` folder.
+2. Press "Add file" and then "Create new file".
+3. Name your file "example.md". Or substitute any other name, as long as it ends with ".md". This extension stands for [Markdown](https://daringfireball.net/projects/markdown/), which will be discussed in more detail below.
+
+## Add the page's "front matter"
```
Comment (jwestw) [inline docs/tutorials/create-pages.md:36]: May want to either define “front matter” immediately or avoid using the term initially.

File: docs/tutorials/create-pages.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+3. In the dropdown select "Create branch: create-pages-tutorial from 'develop'".
+
+This creates a branch called "create-pages-tutorial" and switches to that branch. We will be working on this branch for the rest of the tutorial.
+
+## Start the new page
+
+1. In your site repository on Github.com, go into the `_pages` folder.
+2. Press "Add file" and then "Create new file".
+3. Name your file "example.md". Or substitute any other name, as long as it ends with ".md". This extension stands for [Markdown](https://daringfireball.net/projects/markdown/), which will be discussed in more detail below.
+
```
Comment (jwestw) [inline docs/tutorials/create-pages.md:35]: Potentially add an instruction here: "After writing the filename press enter or click out of the box. The editor is open below in which you can open edit the content of the file."

File: docs/tutorials/create-pages.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+## Topics covered
+
+* Jekyll pages
+* Markdown
+* Multilingual pages
+* Menu configuration
+
+## Level of difficulty
+
+This tutorial is mostly about content, but does involve the editing of some configuration files. Anyone can follow the steps to complete the tutorial, however you will be using the YAML syntax involved in most Open SDG configurations. YAML requires particular indentation and spacing, so some attention to detail will be needed.
```
Comment (jwestw) [inline docs/tutorials/create-pages.md:14]: Would it be a good idea to link to some examples of yaml syntax, [like this](https://blog.stackpath.com/yaml/)? Or would that be over complicating things?

**Review by jwestw — COMMENTED** [review]
Hi @LucyGwilliamAdmin I think I have seen similar errors to this last time I was working on a language branch but I can't remember why this was happening. Please could you take a look at the test output? https://github.com/ONSdigital/sdg-indicators/runs/2166174879?check_suite_focus=true

**Comment by LucyGwilliamAdmin** [comment]
@jwestw re. the error, compare what you've put in the front matter of the new files compared with what's in the front matter of the existing files

**Comment by jwestw** [comment]
@brockfanning 

`custom.my-example.page` is showing up in menu. Do you know the exact term I should use [here](https://github.com/ONSdigital/sdg-indicators/blob/cccf0f24bdfe6d078168ce4ce10ead3bd15a4bf9/_config.yml#L140)? 

image

File: docs/tutorials/create-pages.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+This step is not required, if you would like to add a link to the new page in your main menu, here are the steps:
+
+1. In the root of your site repository on Github.com, make sure you are still in the `create-pages-tutorial` branch. If not, select it from the dropdown on the top left.
+1. Click on `_config.yml`.
+1. Click the pencil icon on the right to begin editing the file.
+1. Look for the `menu` section.
+1. Add an item at the bottom of the list, using the same syntax and indentation as the others:
+
+        - path: my-example-page
```
Comment (jwestw) [inline docs/tutorials/create-pages.md:155]: For the UK site, this had to be `  - path: /my-example-page/` (with slashes).

**Comment by brockfanning** [comment]
@jwestw I added a bit more detail about the translation keys - does it help?

I didn't think the slashes in the path were necessary - did you see some problems in the browser?

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning we've been having issues with slashes for a while - if not like that, we seem to get errors in build:

https://github.com/ONSdigital/sdg-indicators/runs/2166504766?check_suite_focus=true#step:5:8

Although I'm sure I've seen code before that's supposed to deal with that

**Comment by jwestw** [comment]
> @jwestw I added a bit more detail about the translation keys - does it help?
> 
> I didn't think the slashes in the path were necessary - did you see some problems in the browser?

I actually didn't try without, but I know **with** slashes worked. And previously I had a series of failed tests from not including slashes in the permalinks in front matter.

**Comment by jwestw** [comment]
I used `menu.guidance` as the key so I didn't have to create another one. It works.

## Russian

image

##  French

image

**Comment by jwestw** [comment]
Thank you for the extra guidance and the changes you have made so far, @brockfanning. I am happy with this tutorial so will pass it now.

**Review by jwestw — APPROVED** [review]
I think all the changes have been made. Thanks

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin @jwestw We may as well get the slash thing right. Questions I'm wondering about are:
1. Is the preceding slash necessary? Eg, "/my-path"
2. Is the trailing slash necessary? Eg, "my-path/"

My impression of that failing test is that it was due to having trailing slashes in FR and RU but not in EN. But I could be wrong. Anyway I think we should probably pick an approach and stick with it, for both permalinks and menu item paths, for consistency. From your experience it sounds like we should recommend always a preceding _and_ a trailing slash?

**Comment by LucyGwilliamAdmin** [comment]
@jwestw can you investigate the points Brock has made please?

**Comment by jwestw** [comment]
- Test with [proceeding slash only](https://github.com/ONSdigital/sdg-indicators/commit/3f396e406fa4e043fc51dfa22dcbcdc43ec49467): https://github.com/ONSdigital/sdg-indicators/actions/runs/680268927
- Test with [trailing slash only](https://github.com/ONSdigital/sdg-indicators/commit/d1afaa5d6c23476b395932523b4d9491fabd0379): https://github.com/ONSdigital/sdg-indicators/actions/runs/680283826
- Test with [no slashes](https://github.com/ONSdigital/sdg-indicators/commit/33f03d532577d9b0fb1e24929714acb6f74d6b92): https://github.com/ONSdigital/sdg-indicators/actions/runs/680298027
- Test with [mixture of slashes](https://github.com/ONSdigital/sdg-indicators/commit/d02be9975886315bdbd12b92a902171eff95953c) : https://github.com/ONSdigital/sdg-indicators/actions/runs/680303150

**Comment by brockfanning** [comment]
@jwestw It might be helpful to extend the tests to play around with the "permalink" of pages also. My thinking is that there would be four possibilities to test. In each, the same change is made to both...

1. [The "path" for the About page in the menu](https://github.com/ONSdigital/sdg-indicators/blob/develop/_config.yml#L122)
2. And [the "permalink" for the About page itself](https://github.com/ONSdigital/sdg-indicators/blob/develop/_pages/about.md)

The four possibilities would be:

1. No slashes (eg, `path: about` and `permalink: about`)
2. Preceding slash (eg, `path: /about` and `permalink: /about`)
3. Trailing slash (eg, `path: about/` and `permalink: about/`)
4. Both slashes (eg, `path: /about/` and `permalink: /about/`)

It's possible that all four will work equally well, but it would be great to know if any of them *don't* work. Would it be possible to have a feature branch for each? I ask because, even if the tests pass, it still would be nice to try it in a browser.

**Comment by jwestw** [comment]
Hey @brockfanning 

These are the results of the various permutations of slashes. Hopefully my commit messages are clear enough to know which is which.  

image

The links to the two that failed:

https://github.com/ONSdigital/sdg-indicators/commit/0ada36ac25fd2c8d6206a701a8faefda6850d461
https://github.com/ONSdigital/sdg-indicators/commit/80b7a13c87fd34dacc282df6eea012784f030331

**Comment by brockfanning** [comment]
@jwestw Nice! So if I'm interpreting correctly, the trailing slash is always required but the preceding slash is not required?

**Comment by brockfanning** [comment]
Brock to update this PR standardizing around the trailing slash, and look for other spots in the documentation that need changing.

**Pull request merged**